### PR TITLE
 Week 1 SQL Assessment

### PR DESCRIPTION
**Pull request by RudraAbhiram04**

### CODE REVIEW

**Comment by RudraAbhiram04** [comment]
Week 1 SQL Assessment

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+insert into emp  
+values(  
+ 7934, 'MILLER', 'CLERK', 7782,  
+ to_date('23-1-1982','dd-mm-yyyy'),  
+ 1300, null, 10  
+);
+select * from emp;
+
+1.select count(empno) from emp;
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:131]: ok
select count(*) from emp;

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+2.select count(unique(DEPTNO)) from emp;
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:133]: not ok.
select count(*) from dept;

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+3.select * from emp;
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:135]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+4.select unique(DEPTNO) from emp;
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:137]: not ok. asked for all department data

select * from dept;

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+5.select sum(sal) from emp;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:140]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+6.select sum(comm) from emp;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:142]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+7.select JOB,comm from emp where comm!=0;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:144]: not ok. distinct job title required

select distinct job where comm is not null

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+8.select empno,hiredate from emp;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:146]: not ok

select sysdate from dual;

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+9.select avg(sal) from emp;
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:147]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+
+10.select count(empno),DEPTNO from emp group by DEPTNO order by DEPTNO;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:150]: ok, it is usually a practice to capture deptno first and then group funcitons

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+11.select count(empno),sum(sal),DEPTNO from emp group by DEPTNO order by DEPTNO;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:152]: not ok 
Ans: select sum(sal),deptno
 from emp
 group by deptno;

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+12.select count(empno), JOB from emp group by JOB;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:154]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+13.select count(empno),avg(sal),JOB from emp group by JOB;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:156]: ok, employee count not asked

Ans:select job ,avg(sal) as avg_emp
 from emp
 group by job
 order by job;

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+14.select empno,EXTRACT(day from hiredate),extract(month from hiredate),extract(year from hiredate) from emp;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:158]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 15]
```diff
@@ -0,0 +1,194 @@
...
+15.select * from emp order by DEPTNO;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:160]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 16]
```diff
@@ -0,0 +1,194 @@
...
+16.select * from emp order by JOB;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:162]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+17.select * from emp order by SAL DESC;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:164]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 18]
```diff
@@ -0,0 +1,194 @@
...
+18.select * from emp order by DEPTNO ASC,SAL DESC;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:166]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 19]
```diff
@@ -0,0 +1,194 @@
...
+19.select * from emp where length(ename)=6;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:168]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 20]
```diff
@@ -0,0 +1,194 @@
...
+20.select min(sal),max(sal) from emp;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:170]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 21]
```diff
@@ -0,0 +1,194 @@
...
+21.select min(sal),max(sal),avg(sal),sum(sal),count(empno) from emp group by DEPTNO;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:172]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 22]
```diff
@@ -0,0 +1,194 @@
...
+22.select * from emp order by hiredate;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:174]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 23]
```diff
@@ -0,0 +1,194 @@
...
+23.select * from emp order by hiredate DESC FETCH FIRST 1 ROWS ONLY;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:176]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 24]
```diff
@@ -0,0 +1,194 @@
...
+24.select * from emp order by hiredate ASC FETCH FIRST 1 ROWS ONLY;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:178]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 25]
```diff
@@ -0,0 +1,194 @@
...
+25.select * from emp order by EXTRACT(year from hiredate) DESC,DEPTNO ASC;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:180]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 26]
```diff
@@ -0,0 +1,194 @@
...
+26.select * from emp where sal>=(select avg(sal) from emp);
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:182]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 27]
```diff
@@ -0,0 +1,194 @@
...
+27.select * from emp where sal<=(select avg(sal) from emp);
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:184]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 28]
```diff
@@ -0,0 +1,194 @@
...
+28.select * from emp where sal between 2000 and 4000;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:186]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 29]
```diff
@@ -0,0 +1,194 @@
...
+29.select empno,ename,sal from emp where sal=(select min(sal) from emp) or sal=(select max(sal) from emp);
+// To display the salaries of the employees according to the given Question.
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:188]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 30]
```diff
@@ -0,0 +1,194 @@
...
+ select empno,ename,sal from emp where sal=(select min(sal) from emp) or sal=(select max(sal) from emp) order by sal DESC;
+
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:190]: ok

File: Assignments/Rudra/Week 1 SQL Assessment [hunk 31]
```diff
@@ -0,0 +1,194 @@
...
+30. //displaying the details of the employees who had hired in next month.
+select * from emp where EXTRACT(month from hiredate)=EXTRACT(month from add_months(sysdate,1));
+//counting the number of employees.
+select count(empno) from emp where EXTRACT(month from hiredate)=EXTRACT(month from add_months(sysdate,1));
```
Comment (raviproengage) [inline Assignments/Rudra/Week 1 SQL Assessment:194]: ok

**Review by raviproengage — COMMENTED** [review]
Good submission. Keep it up!

**Comment by RudraAbhiram04** [comment]
Thank you Sir. I will correct the mistakes which i had done earlier.

**Pull request closed without merging**